core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -697,8 +697,9 @@ private RemoveCommand createRemoveCommand(Object key, long explicitFlags) {
    }
 
    final CompletableFuture<Void> removeLifespanExpired(K key, V value, Long lifespan, long explicitFlags) {
+      long flagBitSet = explicitFlags | FlagBitSets.SKIP_CACHE_LOAD | FlagBitSets.SKIP_SHARED_CACHE_STORE | FlagBitSets.SKIP_XSITE_BACKUP;
```

Comment:
I don't think we can do this here. We need to set the flag on the cache in the reaper. Otherwise a get which removes the expired entry won't remove the entry from the shared store possibly.